mlir/lib/Dialect/Rock/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -969,6 +969,15 @@ struct GridwiseGemmV2RewritePattern
     std::tie(bVectorDim, bVectorLen) = bestVectorization(
         b, matB, bCopyPerThread, vectorTiebreaker, kPerBlock, nPerBlock);
 
+    // Vectorizing more than the physical vector length (128 bits) might
+    // be harmful for coalescence and other metrics. Let's limit the maximum
+    // amount of data to load to the maximum vector length. This means a
+    // warp will issue, if possible, a global_load_dwordx4 instruction
+    const int64_t maxVectorLenBits = 128;
+    auto bwidth = elementType.getIntOrFloatBitWidth();
+    aVectorLen = std::min(maxVectorLenBits / bwidth, aVectorLen);
+    bVectorLen = std::min(maxVectorLenBits / bwidth, bVectorLen);
```

Comment:
I more think this should be moved inside `bestVectorization()` because we use this routine in a number of cases. So you might get even more perf improvements in this PR.

Reply:
My impression is that 128 is the upper limit of hardware vectorization but I can be wrong.

@krzysz00 Opinion on LDS vectorization size?

Reply:
There is a 128-bit LDS vector load, yeah

(You may not see it because the compiler uses the 2 64-bit loads version when the LDS access is unassigned)

Reply:
I investigated this yesterday, and yes, the register structure is such that 128 can be seen as a hardware constant: more details in https://github.com/ROCmSoftwarePlatform/llvm-project-private/issues/731

Reply:
I'd go ahead and make the move, and that'll make `getMaxVectorization()` take a data type argument like it logically ought to